sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -1428,6 +1428,36 @@ def process(self, element_restriction, *args, **kwargs):
   return _create_sdf_operation(SplitAndSizeRestrictions, *args)
 
 
+@BeamTransformFactory.register_urn(
+    common_urns.sdf_components.TRUNCATE_SIZED_RESTRICTION.urn,
+    beam_runner_api_pb2.ParDoPayload)
+def create_truncate_sized_restriction(*args):
+  class TruncateAndSizeRestriction(beam.DoFn):
+    def __init__(self, fn, restriction_provider, watermark_estimator_provider):
+      self.restriction_provider = restriction_provider
+      self.dofn_signature = common.DoFnSignature(fn)
+
+    def process(self, element_restrictin, *args, **kwargs):
+      ((element, (restriction, estimator_state)), _) = element_restrictin
+      try:
+        truncated_restriciton = self.restriction_provider.truncate(
```

Comment:
We should be returning the same as split_and_size allowing for 0 restrictions to be returned.